test/plugin.js

```diff
@@ -170,6 +170,45 @@ describe('Plugin', function () {
         expect(error).to.equal(reportTitleError);
       });
     });
+
+    describe('compressionAlgorithm', function () {
+      it('should default to gzip', async function () {
+        const config = makeWebpackConfig({
+          analyzerOpts: {}
+        });
+        await webpackCompile(config, '4.44.2');
+        await expectValidReport({
+          parsedSize: 1311,
+          gzipSize: 342
+        });
+      });
+
+      it('should support gzip', async function () {
+        const config = makeWebpackConfig({
+          analyzerOpts: {
+            compressionAlgorithm: 'gzip'
+          }
+        });
+        await webpackCompile(config, '4.44.2');
+        await expectValidReport({
+          parsedSize: 1311,
+          gzipSize: 342
+        });
+      });
+
+      it('should support brotli', async function () {
+        const config = makeWebpackConfig({
+          analyzerOpts: {
+            compressionAlgorithm: 'brotli'
+          }
+        });
+        await webpackCompile(config, '4.44.2');
+        await expectValidReport({
+          parsedSize: 1311,
+          gzipSize: 302
```

Comment:
I think that, even though it increases the size of this PR, it might be better to aim to use `brotliSize` in the data when Brotli is used instead of `gzipSize`. It would be quite confusing for direct users of the report JSON file (e.g. those who supply `--analyzerMode json` value to CLI) if the key would remain as `gzipSize` even though Brotli was selected.

What do you think? Does the confusion risk outweigh changing the code more?

Reply:
Maybe just `compressedSize`?

Reply:
@th0r I thought about this too, as I'm not a friend of key explosion.

On the other hand I'm OK with `gzipSize`, `brotliSize`:
* backwards compatible
* theoretically allows for dual-compression stats